### PR TITLE
Fix Sequel warnings

### PR DESCRIPTION
**Pull request by flash-gordon**
@solnic this requires a thorough look. We definitely have a leaking abstraction regarding qualified SQL names

### CODE REVIEW

File: lib/rom/sql/relation/reading.rb [hunk 1]
```diff
@@ -341,8 +341,10 @@ def where(*args, &block)
             where(*args).where(self.class.schema.restriction(&block))
           elsif args.size == 1 && args[0].is_a?(Hash)
             new(dataset.where(coerce_conditions(args[0])))
-          else
+          elsif !args.empty?
             new(dataset.where(*args))
+          else
+            self
           end
```
Comment (solnic) [inline lib/rom/sql/relation/reading.rb]: hmm, so, now you can call `relation.where` and it just returns self?

**Comment by solnic** [comment]
Do I understand correctly that this will make `where(foo__bar: 'baz')` stop working?

File: lib/rom/sql/relation/reading.rb (hunk 1 above)
Comment (flash-gordon) [inline lib/rom/sql/relation/reading.rb]: ah, `Sequel` doesn't like it anymore, `new(dataset)` is another option here

Comment (solnic) [inline lib/rom/sql/relation/reading.rb]: doesn't like what exactly?

**Comment by flash-gordon** [comment]
@solnic it works, just issues a warning. I think I'll just set `split_symbols = true` for some specs so that we can ensure it works.

File: lib/rom/sql/relation/reading.rb (hunk 1 above)
Comment (flash-gordon) [inline lib/rom/sql/relation/reading.rb]: calling `where/having` without args and a block

Comment (flash-gordon) [inline lib/rom/sql/relation/reading.rb]:
```
SEQUEL DEPRECATION WARNING: Passing no arguments and no block to a filtering method is deprecated and will be removed in Sequel 5.  Include at least one argument or a block when calling a filtering method.
```

**Comment by solnic** [comment]
@flash-gordon yeah it'd be good, we'll get rid of this in 2.0 anyway

**Comment by flash-gordon** [comment]
sure, will do, I forgot to add it's WIP 😆

File: lib/rom/sql/relation/reading.rb (hunk 1 above)
Comment (solnic) [inline lib/rom/sql/relation/reading.rb]: OK so when do we do this?

Comment (flash-gordon) [inline lib/rom/sql/relation/reading.rb]: here https://github.com/rom-rb/rom-sql/blob/c426a83c65b701c8a7eb29052f5c23e29a54e446/lib/rom/sql/relation/reading.rb#L775-L777

**Comment by flash-gordon** [comment]
@solnic works on my app, do we need anything else?

**Comment by flash-gordon** [comment]
I'm merging this in master since I released v1.3.1

**Pull request merged**